tests/test_integration.py

```diff
@@ -1013,12 +1017,85 @@ async def test_info_links(links):
     assert stderr == ""
 
 
+@pytest.mark.asyncio
+async def test_info_remote_data(envsonschema):
+    jsonout, stderr = await bowtie(
+        "info",
+        "--format",
+        "json",
+        "-i",
+        envsonschema,
+        json=True,
+    )
+
+    open_issues_count = 0
+    open_prs_count = 0
+    stars_count = 0
+    watchers_count = 0
+    last_commit_date = ""
+    last_release_date = ""
+
+    try:
```

Comment:
This has a few issues --

It essentially reproduces the code being tested, which is always a bad sign. The point of a test is to ensure some other code is correct. If the test reproduces that code, that's 2 places it could be equally wrong, and that just means when someone figures out it's wrong, they have to change 2 places instead of 1. Somehow a test has to be simpler to see it is correct than the code it tests.

It's certainly good and right to want to test the code you wrote. The real way to do so is generally to be able to inject in a fake HTTP client to the `GitHub` object you're using, so that you can make it *think* it spoke to GitHub but in fact have it simply be given a response from GitHub that you have already retrieved ahead of time. How easy that is I don't recall, but I can have a look and try to give a suggestion in the next few days.

Reply:
> How is it possible to know these values without doing a real API call ?

It's not possible to know the *real* values, but it *is* possible to simply have dummy values, but still in the right "shape".

We don't care whether the watcher count is 3427 to write a test -- we just care that it's a number inside a property called `"watchers_count"` or whatever, and same for all the properties. So if you retrieve one "real response" and simply store it you now have a fake response you can use. I again can spend some time trying to give you an even more specific suggestion if that doesn't help yet, but probably not till the weekend.

Reply:
Ohh okayy. I got your point. Thanks!. I will just retrieve one real response and make the changes.